citest/json_predicate/map_predicate.py

```diff
@@ -153,6 +153,21 @@ def __eq__(self, result):
             and self.__good_map == result.good_object_result_mappings
             and self.__bad_map == result.bad_object_result_mappings)
 
+  def clone_with_source(self, source, base_target_path, base_value_path):
+    """Implements CloneableWithNewSource interface."""
+    builder = MapPredicateResultBuilder(self.pred)
+
+    for index, orig in enumerate(self.results):
+      if isinstance(orig, predicate.CloneableWithNewSource):
```

Comment:
Does python consider an one object an instance of another if it just overrides a function? It seems like there's no explicit object dependency that would cause this to be true.